zio-http/shared/src/main/scala/zio/http/internal/FormState.scala

```diff
@@ -54,8 +56,20 @@ private[http] object FormState {
 
       val crlf = byte == '\n' && !lastByte.isEmpty && lastByte.get == '\r'
 
+      var boundaryDetected = false;
+      val posInLine        = bufferSize + (if (this.lastByte.isEmpty) 0 else 1)
+      if (posInLine < boundary.closingBoundaryBytes.size) {
```

Comment:
Maybe save `boundary.closingBoundaryBytes.size` in a val during the creation of the buffer since it's used in multiple places

Reply:
see b338c6e347ace6ea3cb0bd5d918e3ce4dcdfa16b